pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -86,13 +87,14 @@ var _ CIDRAllocator = (*cloudCIDRAllocator)(nil)
 // NewCloudCIDRAllocator creates a new cloud CIDR allocator.
 func NewCloudCIDRAllocator(client clientset.Interface, cloud cloudprovider.Interface, nodeInformer informers.NodeInformer) (CIDRAllocator, error) {
 	if client == nil {
-		klog.Fatalf("kubeClient is nil when starting NodeController")
+		klog.ErrorS(errors.New("kubeClient is nil when starting cloud CIDR allocator"), "")
```

Comment:
This can be simplified:

```suggestion
		klog.ErrorS(nill, "kubeClient is nil when starting cloud CIDR allocator")
```

I don't know what the code owners prefer: killing the process (as in the original klog.Fatal and now here with klog.FushAndExit) or returning an error.

Let's keep it as it for now because the PR makes sense this way (just changes how logging is done).